File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/housecleaning/HouseCleaningService.java
package com.neatorobotics.sdk.android.robotservices.housecleaning;

import com.neatorobotics.sdk.android.nucleo.RobotConstants;
import com.neatorobotics.sdk.android.robotservices.cleaning.CleaningService;


/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public abstract class HouseCleaningService extends CleaningService {

    public HouseCleaningService() {
        this.cleaningType = RobotConstants.ROBOT_CLEANING_CATEGORY_HOUSE;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/housecleaning/HouseCleaningService.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices.housecleaning

import com.neatorobotics.sdk.android.models.CleaningCategory
import com.neatorobotics.sdk.android.models.Robot
import com.neatorobotics.sdk.android.models.hasService
import com.neatorobotics.sdk.android.robotservices.RobotServices
import com.neatorobotics.sdk.android.robotservices.cleaning.CleaningService

val Robot.houseCleaningService: HouseCleaningService?
    get() {
        if (this.state == null) return null
        return if (this.hasService(RobotServices.SERVICE_HOUSE_CLEANING)) {
            HouseCleaningServiceFactory.get(this.state!!.availableServices[RobotServices.SERVICE_HOUSE_CLEANING]!!)
        } else
            null
    }

abstract class HouseCleaningService : CleaningService() {

    init {
        this.cleaningType = CleaningCategory.HOUSE
    }
}
